src/libsyntax_pos/lib.rs

```diff
@@ -159,6 +159,18 @@ impl Span {
         Span::new(BytePos(lo), BytePos(lo), span.ctxt)
     }
 
+    /// Returns a new span representing the previous character after the start-point of this span
+    pub fn prev_point(self) -> Span {
+        let span = self.data();
+        let span_lo = span.lo.0;
+        let lo = if span_lo == 0 {
+            0
```

Comment:
This looks likely to introduce an infinite loop in *some* case, where we go up to the start of the file. You can return an `Option<Span>` instead.